### ISSUE TITLE
Fix for CA1822 does not respect preferred modifier order

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#5321]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static`

### Analyzer source

**SDK**: 6.0.100-preview.6.21355.2

### Describe the bug

The code fix for [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static` changes the existing modifier order, which introduces a violation of [IDE0036](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0036).

### Steps To Reproduce

```csharp
        protected internal bool EnsureProtocolVersionComplies( // CA1822
```

### Expected behavior

```csharp
        protected internal static bool EnsureProtocolVersionComplies(
```

### Actual behavior

```csharp
        internal protected static bool EnsureProtocolVersionComplies( // IDE0036
```

## Additional context

```
csharp_preferred_modifier_order = public,private,protected,internal,static,extern,new,virtual,abstract,sealed,override,readonly,unsafe,volatile,async:suggestion
```
